### PR TITLE
RD-4149 Use `widget.css` import instead of `hasStyle` flag

### PR DESCRIPTION
**Pull request by kubama**
## Description
Used `widget.css` import instead of `hasStyle` flag. The style becomes part of each widget bundle. This eliminates the need for explicit, separate HTTP call to fetch the CSS.
The `hasStyle` flag is still supported.

## Screenshots / Videos
N/A

## Checklist

- [X] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [X] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [X] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/2684/
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/2685/ - `blueprints_spec` rerun

 

## Documentation
N/A

### CODE REVIEW

**Review by NorbertAmartus — COMMENTED** [review]
Seems nice and clean 🎉 
I wanted to ask about removing support of the `hasStyle` property, but if I can recall correctly, we can not do that due to the fact that users are able to create their own widgets (which may be using this property)

But other than that, 1 question is arising 😁
Would it be a good option, to create a new UI Code Style rule, which would explicitly say that we are avoiding `hasStyle` property in favor of importing `./widget.css` file? 🤔

**Review by NorbertAmartus — APPROVED** [review]
As a comment above is a "non-blocking" one, I'm approving this PR with nothing else than a smile on my face! 🥳

**Review by qooban — APPROVED** [review]
Looks good!

File: widgets/blueprintCatalog/src/widget.tsx [hunk 1]
```diff
@@ -1,8 +1,9 @@
-import RepositoryList from './RepositoryList';
 import Actions from './actions';
 import Consts from './consts';
+import RepositoryList from './RepositoryList';
 
-import type { BlueprintCatalogPayload, BlueprintCatalogWidgetConfiguration, Blueprint } from './types';
+import type { Blueprint, BlueprintCatalogPayload, BlueprintCatalogWidgetConfiguration } from './types';
+import './widget.css';
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/blueprintCatalog/src/widget.tsx:6]: In both approaches (old and new) the CSS is applied globally.

**Comment by kubama** [comment]
> But other than that, 1 question is arising grin Would it be a good option, to create a new UI Code Style rule, which would explicitly say that we are avoiding `hasStyle` property in favor of importing `./widget.css` file? thinking

I'll add a point about that.

**Comment by qooban** [comment]
@kubama @NorbertAmartus 

Regarding the code style update, I don't know if you have considered main [UI Code Style](https://cloudifysource.atlassian.net/wiki/spaces/RD/pages/435388492/UI+Code+Style) page, but probably the best place would be [Adding new built-in widget to Cloudify UI](https://cloudifysource.atlassian.net/wiki/spaces/RD/pages/435421230/Adding+new+built-in+widget+to+Cloudify+UI) (it's referenced from the UI Code Style main page).

**Comment by kubama** [comment]
> Regarding the code style update, I don't know if you have considered main [UI Code Style](https://cloudifysource.atlassian.net/wiki/spaces/RD/pages/435388492/UI+Code+Style) page, but probably the best place would be [Adding new built-in widget to Cloudify UI](https://cloudifysource.atlassian.net/wiki/spaces/RD/pages/435421230/Adding+new+built-in+widget+to+Cloudify+UI) (it's referenced from the UI Code Style main page).

I added the note to [Styling React components](https://cloudifysource.atlassian.net/wiki/spaces/RD/pages/435486834/Styling+React+components) as it was already mentioning `widget.css` file. I'll reference it from [Adding new built-in widget to Cloudify UI](https://cloudifysource.atlassian.net/wiki/spaces/RD/pages/435421230/Adding+new+built-in+widget+to+Cloudify+UI).

**Pull request merged**